lib/pgBackRest/Storage/S3/Driver.pm

```diff
@@ -0,0 +1,512 @@
+            if (index($strPrefix, qw{/}) == -1)
+            {
+                undef($strPrefix);
+            }
+            else
+            {
+                $strPrefix = dirname($strPrefix) . qw{/};
+            }
+        }
+
+        # Store files
+        foreach my $oFile (xmlTagChildren($oResponse, "Contents"))
+        {
+            my $strName = xmlTagText($oFile, "Key");
+
+            # Strip off prefix
+            if (defined($strPrefix))
+            {
+                $strName = substr($strName, length($strPrefix));
+            }
+
+            $hManifest->{$strName}->{type} = 'f';
+            $hManifest->{$strName}->{size} = xmlTagText($oFile, "Size");
+            $iFileTotal++;
+
+            # Generate paths from the name if recursing
+            if ($bRecurse)
+            {
+                my @stryName = split(qw{/}, $strName);
+
+                if (@stryName > 1)
+                {
+                    $strName = undef;
+
+                    for (my $iIndex = 0; $iIndex < @stryName - 1; $iIndex++)
+                    {
+                        $strName .= (defined($strName) ? qw{/} : '') . $stryName[$iIndex];
+                        $hManifest->{$strName}->{type} = 'd';
+                    }
+                }
+            }
+        }
+
+        # Store directories
+        if ($bPath && !$bRecurse)
+        {
+            foreach my $oPath (xmlTagChildren($oResponse, "CommonPrefixes"))
+            {
+                my $strName = xmlTagText($oPath, "Prefix");
+
+                # Strip off prefix
+                if (defined($strPrefix))
+                {
+                    $strName = substr($strName, length($strPrefix));
+                }
+
+                # Strip off final /
+                $strName = substr($strName, 0, length($strName) - 1);
+
+                $hManifest->{$strName}->{type} = 'd';
+                $iPathTotal++;
+            }
+        }
+
+        $strContinuationToken = xmlTagText($oResponse, "NextContinuationToken", false);
+    }
+    while (defined($strContinuationToken));
+
+    # Add . for the initial path (this is just for compatibility with filesystems that have directories)
+    if ($bPath)
+    {
+        $hManifest->{qw{.}}->{type} = 'd';
+    }
+
+    # use Data::Dumper; &log(WARN, 'MANIFEST' . Dumper($hManifest));
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'hManifest', value => $hManifest, trace => true}
+    );
+}
+
+####################################################################################################################################
+# list - list a directory
+####################################################################################################################################
+sub list
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strPath,
+    ) =
```

Comment:
In the Posix drive, bIgnoreMissing is an option and appears to be used (e.g. lines 306, 318 ArchiveCommon.pm, 319, 337, 349 ArchiveInfo.pm, 178 ArchivePush.pm, 242 Expire.pm, 339 Info.pm, etc) so I'm wondering why it is not an option here - is it only because this list function is for the manifest? Shouldn't this function then be called manifestList?